test/functional/email_confirmations_controller_test.rb

```diff
@@ -130,6 +130,97 @@ class EmailConfirmationsControllerTest < ActionController::TestCase
     end
   end
 
+  context "on POST to webauthn_update" do
+    context "with webauthn enabled" do
+      setup do
+        @user = create(:user)
+        @webauthn_credential = create(:webauthn_credential, user: @user)
+        get :update, params: { token: @user.confirmation_token, user_id: @user.id }
+        @challenge = session[:webauthn_authentication]["challenge"]
+        @origin = "http://localhost:3000"
+        @rp_id = URI.parse(@origin).host
+        @client = WebAuthn::FakeClient.new(@origin, encoding: false)
+        WebauthnHelpers.create_credential(
+          webauthn_credential: @webauthn_credential,
+          client: @client
+        )
+        post(
+          :webauthn_update,
+          params: {
+            user_id: @user.id,
+            token: @user.confirmation_token,
+            credentials:
+            WebauthnHelpers.get_result(
+              client: @client,
+              challenge: @challenge
+            )
+          }
+        )
+      end
+
+      should "redirect to root" do
+        assert_redirected_to root_url
+      end
+
+      should "change the user's email" do
+        assert @user.reload.email_confirmed
+      end
+    end
+
+    context "when not providing credentials" do
+      setup do
+        @user = create(:user)
+        @webauthn_credential = create(:webauthn_credential, user: @user)
```

Comment:
Since we don't use these credentials in this context, I'd prefer to not save them to an instance variable. That way it's a bit clearer to the reader that they're unused. (There's also a fairly common convention I've seen of putting an `_` before an unused variable name, but i don't think i've seen it much in this code base so i usually just skip the name.)